docker/DOCKER.md

```diff
@@ -0,0 +1,410 @@
+# Docker Images for the balena CLI
+
+Docker images with the balena CLI and Docker-in-Docker.
+
+## Features Overview
+
+These CLI images are based on the popular [Balena base images](https://www.balena.io/docs/reference/base-images/base-images/)
+so they include many of the features you see there.
+
+- Multiple Architectures:
+    - `rpi`
+    - `armv7hf`
+    - `aarch64` (debian only)
+    - `amd64`
+    - `i386`
+- Multiple Distributions
+    - `debian`
+    - `alpine`
+- [cross-build](https://www.balena.io/docs/reference/base-images/base-images/#building-arm-containers-on-x86-machines) functionality for building ARM containers on x86.
+- Helpful package installer script called `install_packages` inspired by [minideb](https://github.com/bitnami/minideb#why-use-minideb).
+
+Note that there are some additional considerations when running the CLI via Docker so
+pay close attention to the [Usage](#usage) section for examples of different CLI commands.
+
+## Image Names
+
+`balenalib/<arch>-<distro>-balenacli:<cli_ver>`
+
+- `<arch>` is the architecture and is mandatory. If using `Dockerfile.template`, you can replace this with `%%BALENA_ARCH%%`.
+For a list of available device names and architectures, see the [Device types](https://www.balena.io/docs/reference/base-images/devicetypes/).
+- `<distro>` is the Linux distribution and is mandatory. Currently there are 2 distributions, namely `debian` and `alpine`.
+
+## Image Tags
+
+In the tags, all of the fields are optional, and if they are left out, they will default to their `latest` pointer.
+
+- `<cli_ver>` is the version of the balena CLI, for example, `12.40.2`, it can also be substituted for `latest`.
+
+## Examples
+
+`balenalib/amd64-debian-balenacli:12.40.2`
+
+- `<arch>`: amd64 - suitable for running on most workstations
+- `<distro>`: debian - widely used base distro
+- `<cli_ver>`: 12.40.2
+
+`balenalib/armv7hf-alpine-balenacli`
+
+- `<arch>`: armv7hf - suitable for running on a Raspberry Pi 3 for example
+- `<distro>`: alpine - smaller footprint than debian
+- `<cli_ver>`: omitted - the latest available CLI version will be used
+
+## Volumes
+
+Volumes can be used to persist data between instances of the CLI container, or to share
+files between the host and the container.
+In most cases these are optional, but some examples will highlight when volumes are required.
+
+- `-v "balena_data:/root/.balena"`: persist balena credentials and downloads between instances
+- `-v "docker_data:/var/lib/docker"`: persist cache between instances when using Docker-in-Docker (requires `-e "DOCKERD=1"`)
+- `-v "$PWD:$PWD" -w "$PWD"`: bind mount your current working directory in the container to share app sources or balenaOS image files
+- `-v "${SSH_AUTH_SOCK}:/ssh-agent"`: bind mount your host ssh-agent socket with preloaded SSH keys
```

Comment:
This worked for me on Windows, but not on macOS. Maybe related to my system's permissions.
```
$ echo "${SSH_AUTH_SOCK}:/ssh-agent"
/private/tmp/com.apple.launchd.CJQ4XK0H8e/Listeners:/ssh-agent
```
Even when testing with `sudo` and `--privileged`, it would fail (same as when not using `sudo` or `--privileged`):
```
$ sudo docker run -it --privileged -v "${SSH_AUTH_SOCK}:/ssh-agent" -v balena_data:/root/.balena balenalib/amd64-alpine-balenacli:d40af7fdcff1943ffe488c3067400dbb57f8645d /bin/bash

bash-5.0# ssh-add -l
Error connecting to agent: Connection refused

bash-5.0# env
SSH_AUTH_SOCK=/ssh-agent
...

bash-5.0# ls -la /ssh-agent
srw-rw-rw-    1 root     root             0 Feb 15 01:58 /ssh-agent
```
But on the "host OS" (my laptop):

```
$ env | grep SSH
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.CJQ4XK0H8e/Listeners

$ ls -la /private/tmp/com.apple.launchd.CJQ4XK0H8e/Listeners
srw-rw-rw-  1 paulo  wheel  0 15 Feb 01:58 /private/tmp/com.apple.launchd.CJQ4XK0H8e/Listeners

$ ssh-add -l
2048 SHA256:0XZ70WNycHiGhmVsGjqK2ajiuUvRPG1xoUaI/J/s2Uw /Users/paulo/.ssh/id_rsa (RSA)
```

We don't need to change DOCKER.md, unless you have suggestions.

Reply:
I don't have a Mac to test on, but is it possible that `/private/tmp/` isn't shared with Docker Desktop by default?